pallets/creator-staking/src/types/mod.rs

```diff
@@ -0,0 +1,202 @@
+use codec::{Decode, Encode, HasCompact, MaxEncodedLen};
+use frame_support::traits::Currency;
+use scale_info::TypeInfo;
+use sp_arithmetic::traits::AtLeast32BitUnsigned;
+use sp_runtime::{Perbill, traits::Zero, RuntimeDebug};
+use sp_std::{ops::Add, prelude::*};
+
+use super::*;
+
+pub mod impls;
+
+pub(crate) type BalanceOf<T> =
+<<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+/// Counter for the number of eras that have passed.
+pub type EraIndex = u32;
+
+// This represents the max assumed vector length that any storage item should have.
+// In particular, this relates to `UnbondingInfo` and `StakerInfo`.
+// In structs which are bound in size, `MaxEncodedLen` can just be derived but that's not the case
+// for standard `vec`. To fix this 100% correctly, we'd need to do one of the following:
+//
+// - Use `BoundedVec` instead of `Vec` and do storage migration
```

Comment:
They are relevant, because we still use custom implementation of `MaxEncodedLen`. There's a way to improve it.
More here: https://github.com/dappforce/subsocial-parachain/pull/222#discussion_r1315661932